File: live-test/javascript/src/normal.js
'use strict';

class Drawable {
    draw() { throw new Error('not implemented'); }
}

class Shape {
    area() { throw new Error('abstract'); }
    describe() { return `${this.constructor.name} with area ${this.area()}`; }
}

class Circle extends Shape {
    constructor(radius) {
        super();
        this.radius = radius;
    }
    area() { return 3.14159 * this.radius * this.radius; }
    draw() { return `circle r=${this.radius}`; }
}

class Rectangle extends Shape {
    constructor(width, height) {
        super();
        this.width = width;
        this.height = height;
    }
    area() { return this.width * this.height; }
    draw() { return `rect ${this.width}x${this.height}`; }
}

class Square extends Rectangle {
    constructor(side) {
        super(side, side);
    }
}

class ShapeCollection {
    constructor() {
        this.shapes = [];
    }
    add(shape) { this.shapes.push(shape); }
    totalArea() {
        let sum = 0;
        for (const s of this.shapes) sum += s.area();
        return sum;
    }
    largest() {
        let best = null;
        for (const s of this.shapes) {
            if (best === null || s.area() > best.area()) best = s;
        }
        return best;
    }
}

function makeDefaultShapes() {
    return [new Circle(1.0), new Rectangle(2.0, 3.0), new Square(4.0)];
}

module.exports = { Drawable, Shape, Circle, Rectangle, Square, ShapeCollection, makeDefaultShapes };

File: live-test/python/src/normal.py
"""Vanilla OOP patterns for the live MCP test harness."""
from abc import ABC, abstractmethod
from typing import Protocol


class Drawable(Protocol):
    def draw(self) -> str: ...


class Shape(ABC):
    @abstractmethod
    def area(self) -> float:
        ...

    def describe(self) -> str:
        return f"{type(self).__name__} with area {self.area()}"


class Circle(Shape):
    def __init__(self, radius: float) -> None:
        self.radius = radius

    def area(self) -> float:
        return 3.14159 * self.radius * self.radius

    def draw(self) -> str:
        return f"circle r={self.radius}"


class Rectangle(Shape):
    def __init__(self, width: float, height: float) -> None:
        self.width = width
        self.height = height

    def area(self) -> float:
        return self.width * self.height

    def draw(self) -> str:
        return f"rect {self.width}x{self.height}"


class Square(Rectangle):
    def __init__(self, side: float) -> None:
        super().__init__(side, side)


class ShapeCollection:
    def __init__(self) -> None:
        self.shapes: list[Shape] = []

    def add(self, shape: Shape) -> None:
        self.shapes.append(shape)

    def total_area(self) -> float:
        return sum(s.area() for s in self.shapes)

    def largest(self) -> Shape | None:
        if not self.shapes:
            return None
        return max(self.shapes, key=lambda s: s.area())


def make_default_shapes() -> list[Shape]:
    return [Circle(1.0), Rectangle(2.0, 3.0), Square(4.0)]
